src/web/plugin-admin.jsp

```diff
@@ -106,7 +106,14 @@
             for (Object objItem : items) {
                 FileItem item = (FileItem)objItem;
                 String fileName = item.getName();
+                
                 if (fileName != null) {
+                    // Ensure we're dealing with a jar file.
+                    if (!"application/java-archive".equals(item.getContentType())) {
```

Comment:
Openfire accepts not only `.jar`, but also `.war` files.

The `FileItem` instance is a representation of data in a HTTP request, right? I'm not sure if checking the content type will bring any benefit. A malicious user can easily forge that.

Reply:
Ah, perhaps not so much with the CSRF checks that we now have. Does this make it fully safe? If not, I'm worried that this might introduce more issues than solve any.

Reply:
Isn't the content-type supplied by the browser? Can you be sure that all (reasonable) browsers will set the content-type to exactly application/java-archive and not application/jar or something else?